tests/test2/test2.py

```diff
@@ -0,0 +1,95 @@
+import numpy
+
+from src.geofabrics import lidar_fetch
+from src.geofabrics import geometry
+
+class Test2(unittest.TestCase):
+    """ A class to test the basic dem generation pipeline for a simple example
+    with land, offshore, a reference DEM and lidar using the data specified in 
+    the test1/instruction.json """
+    
+    DATASET = "Wellington_2013"
+    FILES = ["ot_CL1_WLG_2013_1km_085033.laz", "ot_CL1_WLG_2013_1km_086033.laz", 
+             "ot_CL1_WLG_2013_1km_085032.laz", "ot_CL1_WLG_2013_1km_086032.laz", 
+             DATASET + "_TileIndex.zip"]
+    SIZES = [ 6795072, 5712485, 1670549, 72787, 598532]
+
+    @classmethod
+    def setUpClass(cls):
+        
+        # load in the test instructions
+        file_path = pathlib.Path().cwd() / pathlib.Path("tests/test2/instruction.json")
+        with open(file_path, 'r') as file_pointer:
+            instructions = json.load(file_pointer)
+            
+        # define cache location - and catchment dirs
+        cls.cache_dir = pathlib.Path(instructions['instructions']['data_paths']['local_cache'])
+        
+        # ensure the cache directory doesn't exist - i.e. clean up from last test occured correctly
+        if cls.cache_dir.exists():
+            shutil.rmtree(cls.cache_dir)
+        cls.cache_dir.mkdir()
+        # create fake catchment boundary
+        x0 = 1764410; y0 = 5470382; x1 = 1765656; y1 = 5471702;
+        catchment = shapely.geometry.Polygon([(x0, y0), (x1, y0), (x1, y1), (x0, y1)])
+        catchment = geopandas.GeoSeries([catchment])
+        catchment = catchment.set_crs(instructions['instructions']['projection'])
+        
+        # save faked catchment boundary
+        catchment_dir = cls.cache_dir / "catchment"
+        catchment.to_file(catchment_dir)
+        shutil.make_archive(base_name=catchment_dir, format='zip', root_dir=catchment_dir)
+        shutil.rmtree(catchment_dir)
+        
+        # create a catchment_geometry
+        catchment_dir = pathlib.Path(str(catchment_dir) + ".zip")
+        cls.catchment_geometry = geometry.CatchmentGeometry(catchment_dir, catchment_dir,  # all land
+                                                            instructions['instructions']['projection'],
+                                                            instructions['instructions']['grid_params']['resolution'])
+        
+    @classmethod
+    def tearDownClass(cls):
+        if cls.cache_dir.exists():
+            shutil.rmtree(cls.cache_dir)
+
+    def test_download_all(self):
+        """ A test to see if all expected files are downloaded - check files and size """
+        
+        # Run pipeline
+        runner = lidar_fetch.OpenTopography(self.catchment_geometry, self.cache_dir, verbose = True)
+        runner.run()
+        
+        # check the right dataset is downloaded - self.DATASET
+        assert len(list(self.cache_dir.glob('*/**'))) == 1, "There should only be one dataset named " + self.DATASET + " instead there are " + \
```

Comment:
using unittest assert statements would allow better integration with test runners and IDEs.

`self.assertEquals(actual, expected, msg)`, `self.assertTrue(bool, msg)` etc

Reply:
good suggestion. I've made it!

Reply:
Next i'll tidy up the messages with string formatting as per your earlier comment.